### ISSUE TITLE
elementsToString and asCharSequence for CharLists

### ISSUE THREAD

**Issue by techsy730** [vigna/fastutil#224]
It may be useful to have `elementsToString` (returns a new String with the elements of the CharCollection as its characters) and `asCharSequence` (presents a view of the `CharList` as a `CharSequence`, no copying) in `CharList`. Perhaps `elementsToString` would make sense in `CharCollection` too as there are some ordered non-list collections where it might make sense (SortedSet, LinkedHashSet, etc)

I'm not sure if `elementsToString` is really a good name. `contentsToString` and `stringValue` also come to mind.
I do like `asCharSequence` though.

**Comment by techsy730** [comment]
I've got a PR in progress but I would like to source some feedback before I go too much further with it.

**Comment by vigna** [comment]
The conversion to string sounds more like a helper method for CharCollections. For the view, I'd use charSequenceView(), which makes clear that no copying is performed.

**Comment by techsy730** [comment]
> The conversion to string sounds more like a helper method for CharCollections.
Yes, but then things like `CharArrayList` won't be able to avoid an extra copy converting to a `String`. We have to copy once into an accumulating array, and then once again done by `String` construction. By letting array based subclasses to provide their own implementation, one copy step can be removed. We can pass the array straight to String (with offset and length) which will make its own copy, but now we are down to one copy instead of two.

I could special case `ArrayList` and `ArraySet` in the static method, but that seems dirty. :tongue:

> For the view, I'd use charSequenceView(), which makes clear that no copying is performed.

Yeah, that sounds good

**Comment by vigna** [comment]
Good point, if you can do much better than a method might be a good idea. Maybe you can default it in CharCollection and then have improved overrides.

Presently I'm more focused on reviewing the code and #162 , as I want either to complete it (with functions) or at least be sure we won't get into trouble in the future when we complete it (e.g., method calls becoming ambiguous).

**Comment by techsy730** [comment]
Thoughts on the name of the method? `elementsToString`, `contentsToString`, or `stringValue`?

And yeah, I wouldn't mind if this even gets pushed to 8.6.0, not a high priority feature by any means.

**Comment by vigna** [comment]
How does `asString()` sound?

**Comment by techsy730** [comment]
`asString` sounds good. Easy enough to remember what the difference is between that and `toString` once you learn about it. We would need some good documentation on the interface's `asString` and `toString` methods what the difference is (the former is the contents of the list into a string, the latter is a representation of the state of the list, usually the contents represented as a list type format like `[a, b, c, ...]`)

**Comment by techsy730** [comment]
Just a heads up, I have a branch halfway done implementing this, but IRL work got busy before I could clean it up for submission

**Comment by techsy730** [comment]
Ok, I have the type I want to expose for unmodifiable view (basically a plain char sequence but also supporting many of the non-mutation methods StringBuilder provides, like `getChars`)
The issue now comes from how many mutation based methods I want to provide, like `java.lang.Appendable`.

I may limit my scope to an unmodifiable view first, and then add in optional mutable views in later PRs
